uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/services/leave-popup.service.js

```diff
@@ -47,9 +47,10 @@ define([
      * @param {String} leaveType
      * @param {String} selectedContactId - Contact ID for the contact dropdown
      *                                     when the manager/admin is opening the request
-     * @param {Boolean} forceRecalculateBalanceChange optional
+     * @param {Boolean} [forceRecalculateBalanceChange]
+     * @param {String} [defaultStatus] - "approved", "rejected" etc.
      */
-    function openModal (leaveRequest, leaveType, selectedContactId, forceRecalculateBalanceChange) {
+    function openModal (leaveRequest, leaveType, selectedContactId, forceRecalculateBalanceChange, defaultStatus) {
```

Comment:
I don't enjoy the path this method is taking. It has too many parameters already and we are adding a new one.

Do you think it's possible, if time allows it, to fix this method? I think there are some things to investigate here:

* Do we really need the `leaveType` and `selectedContactId` parameters? Can't those be inferred by the `leaveRequest`'s object? (`leaveRequest.type_id` and `leaveRequest.contact_id`).
* Boolean parameters are really tricky. The problem is that they don't usually provide much context. If you see this implementation somewhere, what can you infer about the `true` parameter?
```js
leavePopup.openModal({id: 1, type_id: 3, contact_id: 5}, true);
```
I think we should use an "options" parameter instead which includes the `forceRecalculateBalanceChange` and `defaultStatus`:
```js
leavePopup.openModal(
  {id: 1, type_id: 3, contact_id: 5},
  {forceRecalculateBalanceChange: true, defaultStatus: 'approved'}
);

function openModal (leaveRequest, options) {
  // ...
}
```

Reply:
Done. I refactored the params.